code/ai/aicode.cpp

```diff
@@ -15201,118 +15206,125 @@ void maybe_process_friendly_hit(object *objp_hitter, object *objp_hit, object *o
 		return;
 	}
 
-	if ((obj_team(objp_hitter) == obj_team(objp_hit)) && (objp_hitter == Player_obj)) {
+	// Cyborg -- I had to switch around the logic here because it was potentially
+	// sending non-ships to obj_team, which would trigger an assert
+	if (objp_hitter == Player_obj){
 
-		// AL 12-4-97: It is possible the Player is a OBJ_GHOST at this point.  If so, bail out.
+		// Cyborg - So check if this is a ship first. Probably true, but you never know, with FSO
+		// (especially in multi)
 		if ( objp_hitter->type != OBJ_SHIP ) {
 			return;
 		}
```

Comment:
Is it expected that this function may get called on a non-ship object? If Or would that indicate a bug?

Reply:
This is in a section where we are only working with the player ship.  AFAIK, In multi, `Player_obj` can be a ship, observer, or none, depending on the player status.  If the `Player_obj` is some other kind of object, then I would expect we would experience problems everywhere.  If the hitter here is the player and they player is not in their ship anymore, the AI shouldn't really act on that information anyway.